azure-pipelines.yml

```diff
@@ -0,0 +1,419 @@
+            maven
+          path: $(MAVEN_CACHE_FOLDER)
+
+      - task: DownloadSecureFile@1
+        displayName: 'Download Maven settings'
+        name: mavenSettings
+        inputs:
+          secureFile: 'maven-settings.xml'
+
+      - template: update-maven-version-steps.yml
+        parameters:
+          mavenSettingsFilePath: $(mavenSettings.secureFilePath)
+
+      - task: Maven@3
+        displayName: 'Run Maven deploy'
+        env:
+          ARTIFACTORY_DEPLOY_USERNAME: $(ARTIFACTORY_DEPLOY_USERNAME)
+          ARTIFACTORY_DEPLOY_PASSWORD: $(ARTIFACTORY_DEPLOY_PASSWORD)
+        inputs:
+          goals: 'deploy'
+          options: >-
+            $(commonMavenArguments)
+            --settings $(mavenSettings.secureFilePath)
+          publishJUnitResults: false
+          javaHomeOption: 'JDKVersion'
+          jdkVersionOption: '1.11'
+          mavenOptions: $(MAVEN_OPTS)
+
+
+    - job: runJavaUnitTests
+      displayName: 'Run Java Unit Tests'
+      dependsOn: runJavaBuild
+
+      steps:
+
+      - task: DownloadSecureFile@1
+        displayName: 'Download Maven settings'
+        name: mavenSettings
+        inputs:
+          secureFile: 'maven-settings.xml'
+
+      - template: update-maven-version-steps.yml
+        parameters:
+          mavenSettingsFilePath: $(mavenSettings.secureFilePath)
+
+      - task: DownloadPipelineArtifact@2
+        displayName: 'Download .Net binaries for Maven build'
+        inputs:
+          artifact: Binaries
+          targetPath: 'sonaranalyzer-dotnet/src/SonarAnalyzer.Vsix/bin/$(buildConfiguration)/'
+          
+      - task: DownloadPipelineArtifact@2
+        displayName: 'Download rule descriptor binaries'
+        inputs:
+          artifact: RuleDescriptorBin
+          targetPath: 'sonaranalyzer-dotnet\src\SonarAnalyzer.RuleDescriptorGenerator\bin'
+
+      - task: Cache@2
+        displayName: "Cache Maven local repo"
+        inputs:
+          key: 'maven | "$(Agent.OS)" | **/pom.xml'
+          restoreKeys: |
+            maven | "$(Agent.OS)"
+            maven
+          path: $(MAVEN_CACHE_FOLDER)
+
+      - task: SonarCloudPrepare@1
+        displayName: 'Prepare code analysis for Java plugin'
+        inputs:
+          SonarCloud: 'SonarCloud'
+          organization: 'sonarsource'
+          scannerMode: 'Other'
+
+      - task: Maven@3
+        displayName: 'Run Maven verify, UTs with SonarCloud'
+        inputs:
+          goals: 'verify'
+          options: -B --settings $(mavenSettings.secureFilePath) -Pcoverage
+          publishJUnitResults: true
+          testResultsFiles: '**/surefire-reports/TEST-*.xml'
+          testRunTitle: 'UTs on Windows'
+          javaHomeOption: 'JDKVersion'
+          jdkVersionOption: '1.11'
+          mavenOptions: $(MAVEN_OPTS)
+          sonarQubeRunAnalysis: true
+          sqMavenPluginVersionChoice: 'latest'
+
+      - task: SonarCloudPublish@1
+        displayName: 'Run Code Analysis - Publish QG'
```

Comment:
We have 2 `SonarCloudPublish` in the yml file. From our discussion, I get that we mostly care about .NET SonarAnalyzer QG, however, are we sure that it is always the correct QG that will be published to github ?

Reply:
Very good question. Currently only one QG is displayed on GitHub which is the one which completes the last (it overwrites all the previous ones). This is a SonarCloud/Qube limitation.

Since the .Net unit tests finish last (with about 4 minutes difference) they will be the one displayed. This is a happy coincidence since they are the one we want to display.

Reply:
I think we should have a ticket to track this.
Do you know if the sonarcloud team is actively working to fix this?